src/solvers/dg_triangles/mesh.jl

```diff
@@ -0,0 +1,37 @@
+
+abstract type AbstractMeshData{Dim} end
+
+"""
+VertexMappedMesh describes a mesh which is constructed by an reference-to-physical 
+mapping which can be constructed using only the vertex positions. 
+
+Wraps `MeshData` and `boundary_faces` in a dispatchable mesh type.
+"""
+struct VertexMappedMesh{Dim,Nboundaries,Tv,Ti} <: AbstractMeshData{Dim}
+  md::MeshData{Dim,Tv,Ti}
+  boundary_faces::Dict{Symbol,Vector{Int}}
+end
+
+Trixi.ndims(::VertexMappedMesh{Dim}) where {Dim} = Dim
+
+"""
+  function VertexMappedMesh(VXYZ::NTuple{Dim,Vector{Tv}},EToV,rd::RefElemData;
+                            is_on_boundary=nothing,
+                            is_periodic::NTuple{Dim,Bool}=ntuple(_->false,Dim)) where {Dim,Tv}
+
+- `VXYZ` is a tuple of vectors of vertex coordinates
+- `EToV` is a matrix containing element-to-vertex connectivities for each element
+- `is_on_boundary` specifies boundary using Dict{Symbol,<:Function}
+- `is_periodic` is a tuple of booleans specifying periodicity = true/false in the (x,y,z) direction.
+"""
```

Comment:
You can remove the `function` part from the docstring and just state the signature